src/ray/core_worker/context.cc

```diff
@@ -0,0 +1,76 @@
+
+#include "context.h"
+
+namespace ray {
+
+/// per-thread context for core worker.
+struct WorkerThreadContext {
+  WorkerThreadContext()
+      : current_task_id(TaskID::FromRandom()), task_index(0), put_index(0) {}
+
+  int GetNextTaskIndex() { return ++task_index; }
+
+  int GetNextPutIndex() { return ++put_index; }
+
+  const TaskID &GetCurrentTaskID() { return current_task_id; }
+
+  void SetCurrentTask(const raylet::TaskSpecification &spec) {
+    current_task_id = spec.TaskId();
+    task_index = 0;
+    put_index = 0;
+  }
+
+  /// The task ID for current task.
+  TaskID current_task_id;
```

Comment:
I think we should declare these member variables as `private` since we have set/got them through `getter/setter`.